lib/src/progress_bar.dart

```diff
@@ -57,75 +57,87 @@ class _VideoProgressBarState extends State<VideoProgressBar> {
-        final shouldSeekToRelativePosition =
-            !Platform.isAndroid || !controller.value.isBuffering;
-        if (shouldSeekToRelativePosition) {
-          _seekToRelativePosition(details.globalPosition);
-        }
-
-        widget.onDragUpdate?.call();
-      },
-      onHorizontalDragEnd: (DragEndDetails details) {
-        if (_controllerWasPlaying) {
-          controller.play();
-        }
-
-        widget.onDragEnd?.call();
-      },
-      onTapDown: (TapDownDetails details) {
-        if (!controller.value.isInitialized) {
-          return;
-        }
-        _seekToRelativePosition(details.globalPosition);
-      },
-      child: Center(
+    final ChewieController chewieController = ChewieController.of(context);
+
+    return chewieController.draggableProgressBar
+        ? GestureDetector(
+            onHorizontalDragStart: (DragStartDetails details) {
+              if (!controller.value.isInitialized) {
+                return;
+              }
+              _controllerWasPlaying = controller.value.isPlaying;
+              if (_controllerWasPlaying) {
+                controller.pause();
+              }
+
+              widget.onDragStart?.call();
+            },
+            onHorizontalDragUpdate: (DragUpdateDetails details) {
+              if (!controller.value.isInitialized) {
+                return;
+              }
+              _latestDraggableOffset = details.globalPosition;
+              listener();
+
+              widget.onDragUpdate?.call();
+            },
+            onHorizontalDragEnd: (DragEndDetails details) {
+              if (_controllerWasPlaying) {
+                controller.play();
+              }
+
+              if (_latestDraggableOffset != null) {
+                _seekToRelativePosition(_latestDraggableOffset!);
+                _latestDraggableOffset = null;
+              }
+
+              widget.onDragEnd?.call();
+            },
+            onTapDown: (TapDownDetails details) {
+              if (!controller.value.isInitialized) {
+                return;
+              }
+              _seekToRelativePosition(details.globalPosition);
+            },
+            child: progressBar,
+          )
+        : progressBar;
+  }
+
+  Widget get progressBar => Center(
         child: Container(
           height: MediaQuery.of(context).size.height,
           width: MediaQuery.of(context).size.width,
           color: Colors.transparent,
           child: CustomPaint(
             painter: _ProgressBarPainter(
               value: controller.value,
+              draggableValue: _latestDraggableOffset != null
+                  ? _calcRelativePosition(_latestDraggableOffset!)
+                  : Duration.zero,
               colors: widget.colors,
               barHeight: widget.barHeight,
               handleHeight: widget.handleHeight,
               drawShadow: widget.drawShadow,
             ),
           ),
         ),
-      ),
-    );
-  }
+      );
```

Comment:
what is own standalone Widget？

Reply:
Basically, take the resulting code, and spin it off into its own `StatelessWidget`.

Typically, it's an anti-pattern for functions to return `Widget`s. You can get around it by creating a `StatelessWidget` that holds all of that Widget code within its `build` method.

Reply:
Does such a small one need to be extracted?

Reply:
I would like to eventually be able to write tests for this `Widget`, as well as all of the other widgets in this library. 

Size doesn't matter as long as it can be tested in isolation.